exchanges/orderbook/depth.go

```diff
@@ -50,7 +59,7 @@ func NewDepth(id uuid.UUID) *Depth {
 
 // Publish alerts any subscribed routines using a dispatch mux
 func (d *Depth) Publish() {
-	if err := d.mux.Publish(d, d._ID); err != nil {
+	if err := d.mux.Publish(Outbound(d), d._ID); err != nil {
```

Comment:
I still don't get this. Isn't the point of the dispatcher to push updates? You're sending something with a `Retreive()` function, circumventing the process. Why not just use alert then? The orderbook stream in rpcserver doesn't even use the dispatcher and can use it when its disabled because it just loops on using `Retreive()`

If I'm subscribing to get update data for an orderbook, this way of doing things means I can no longer actually see what the actual update was. I can call retreive, but there's no way of knowing that what I'm seeing is the actual update

Reply:
The endpoint using the dispatcher is more of a macro overview for that RPC stream `GetExchangeOrderbookStream` for all the orderbooks on that specific exchange. 

The goal is to do as less processing in this routine in websocket buffer as possible and release it back to the `exch.Websocket.Conn.ReadMessage()` if we were to use the alert system we would have to individually get each *orderbook.Depth supported then call wait on N*enabled_pairs. Scale up and scale down on enabled pairs. 

> If I'm subscribing to get update data for an orderbook, this way of doing things means I can no longer see what the actual update was. I can call retrieve, but there's no way of knowing that what I'm seeing is the actual update.

It's also compounded a little bit worse by the fact that in this update it will not wait for a receiver anymore . The updates themselves are snapshots and there are going to be dropped updates, but you are still getting the most updated version for your strategy to act on in an RPC sense.

Reply:
Thanks for the explanation. I remember that this was one of the reasons for this PR again. I'll have a bit more of a think on this as I agree with your reasoning

Reply:
I think you've addressed my concern regarding the actual orderbook through the interface use, the dispatch works and has the improvements you talk about regarding websocket. Given that as I said, the `orderbook.Depth` and `Alert` systems work, so if someone wanted to build something to monitor each ob update for whatever reason, they still have the ability to do so. So I'll resolve this, thanks again for the explanation 🤙